contrib/containerized-pbench/pbench

```diff
@@ -1,64 +1,57 @@
+# defaults to /var/tmp/{USER}/pbench-agent/run, and maps that directory into
+# the container so that multiple runs can be generated and uploaded at once.
+#
+# To upload results to a Pbench Server, use this script to execute the
+# pbench-results-move command within the container, specifying either --relay
+# with the address of a Pbench Relay Server, or --server with the address of a
+# Pbench Server and --token to specify a Pbench Server API key for user
+# authentication.
+#
+# To use a server with a certificate signed by the Pbench development CA bundle
+# define the environment variable PB_AGENT_CA to cause the CA to be mapped into
+# the container and defined using REQUESTS_CA_BUNDLE:
+#
+#  PB_AGENT_CA=server/pbenchinacan/etc/pki/tls/certs/pbench_CA.crt \
+#    contrib/containerized-pbench/pbench pbench-results-move \
+#      --server https://<server>:8443 --token <api-token>
 
 image_name=${PB_AGENT_IMAGE_NAME:-quay.io/pbench/pbench-agent-all-centos-8:main}
-config_file=${_PBENCH_AGENT_CONFIG:-${HOME}/.config/pbench/pbench-agent.cfg}
 pbench_run_dir=${PB_AGENT_RUN_DIR:-/var/tmp/${USER}/pbench-agent/run}
-pbench_server=${PB_AGENT_SERVER_LOC}
+ca=${PB_AGENT_CA:-${REQUESTS_CA_BUNDLE}}
+if [[ ${ca} ]]; then
+    pbench_ca=$(realpath ${ca})  # expand path outside container
+fi
+container_ca=/etc/pki/tls/certs/pbench_CA.crt  # path inside container
 other_options=${PB_AGENT_PODMAN_OPTIONS}
 
 if [[ $# == 0 || $1 == "help" || $1 == "-h" || $1 == "--help" ]]; then
     echo "Usage:  ${0} <Pbench Agent Command> [<arg>...]" >&2
     exit 2
 fi
 
-if [[ -d "${HOME}/.ssh" && -r "${HOME}/.ssh" ]]; then
-    other_options="--security-opt=label=disable -v ${HOME}/.ssh:/root/.ssh ${other_options}"
-fi
-
-if [[ -f "${config_file}" && -r "${config_file}" ]]; then
-    other_options="-v ${config_file}:/opt/pbench-agent/config/pbench-agent.cfg:z ${other_options}"
-elif [[ -n "${pbench_server}" ]]; then
-    echo "Warning:  the Pbench Agent config file is missing; attempting to generate one in ${config_file}" >&2
-    # TODO:  this should be handled by a separate Pbench Agent "configuration wizard".
-    mkdir -p $(dirname ${config_file})
-    cat > ${config_file} <<- EOF
-		[DEFAULT]
-		pbench_install_dir = /opt/pbench-agent
-		pbench_web_server = ${pbench_server}
-		[config]
-		path = %(pbench_install_dir)s/config
-		files = pbench-agent-default.cfg
-		EOF
-else
-    echo "Warning:  the Pbench Agent config file (e.g., ${config_file}) is missing or inaccessible -- using default configuration." >&2
-fi
-
 mkdir -p ${pbench_run_dir}
-other_options="-v ${pbench_run_dir}:/var/lib/pbench-agent:z ${other_options}"
+if [[ -f "${pbench_ca}" ]]; then
+    other_options="-v ${pbench_ca}:${container_ca}:z ${other_options}"
+    other_options="-e REQUESTS_CA_BUNDLE=${container_ca} ${other_options}"
+fi
+other_options="--security-opt=label=disable -v ${pbench_run_dir}:/var/lib/pbench-agent:z ${other_options}"
```

Comment:
Lines 49 and 54 sort of "go together", so it would be better if the block at line 50 followed line 54 (or preceded line 49, if you like that better).

Reply:
Actually, since it's unconditional, bundling it into `other_options` in the first place was kinda silly ...